pkg/volume/emptydir/empty_dir.go

```diff
@@ -294,12 +294,19 @@ func (ed *emptyDir) assignQuota(dir string, mounterSize *resource.Quantity) erro
 	if mounterSize != nil {
 		// Deliberately shadow the outer use of err as noted
 		// above.
-		hasQuotas, err := fsquota.SupportsQuotas(ed.mounter, dir)
+		// hostUsers field is interpreted as true by default, so a pod is by default
+		// not confined by a user namespace.
+		userNamespacesEnabled := false
+		if ed.pod.Spec.HostUsers != nil {
+			userNamespacesEnabled = !*ed.pod.Spec.HostUsers
+		}
+		hasQuotas, err := fsquota.SupportsQuotas(ed.mounter, dir, userNamespacesEnabled)
+		klog.V(3).Infof("assignQuota called, hasQuotas = %t userNamespacesEnabled = %t", hasQuotas, userNamespacesEnabled)
```

Comment:
wasn't there some feature-gate that also should be checked?

Reply:
yes good point. KAS Won't pass down the HostUsers field if the feature gate isn't enabled there, but it could be enabled in KAS and not kubelet (odd situation) so we should probably catch that

Reply:
> it could be enabled in KAS and not kubelet (odd situation)

It's not odd, it's a common case when there is a version skew between KAS and kubelet.

Reply:
Added a check to see if the userNamespacesSupport feature-gate is enabled